tests/test_wsi_registration.py

```diff
@@ -1,7 +1,104 @@
+import pathlib
+
 import numpy as np
 import pytest
 
-from tiatoolbox.tools.registration.wsi_registration import match_histograms
+from tiatoolbox.tools.registration.wsi_registration import (
+    match_histograms,
+    prealignment,
+)
+from tiatoolbox.utils.misc import imread
+
+
+def test_prealignment(fixed_image, moving_image, fixed_mask, moving_mask):
+    """Test for prealignment of an image pair"""
+    fixed_img = imread(pathlib.Path(fixed_image))
+    moving_img = imread(pathlib.Path(moving_image))
+    fixed_mask = imread(pathlib.Path(fixed_mask))
+    moving_mask = imread(pathlib.Path(moving_mask))
+
+    with pytest.raises(
+        ValueError, match=r".*The input images should be grayscale images.*"
+    ):
+        _ = prealignment(fixed_img, moving_img, fixed_mask, moving_mask)
+
+    expected = np.array([[1, 0, 337.8], [0, 0, 767.7], [0, 0, 1]])
+    fixed_img, moving_img = fixed_img[:, :, 0], moving_img[:, :, 0]
+    transform = prealignment(
+        fixed_img,
+        moving_img,
+        fixed_mask,
+        moving_mask,
+        dice_overlap=0.5,
+        rotation_step=10,
+    )
+    assert transform.shape == (3, 3)
+    assert np.mean((expected[:2, :2] - transform[:2, :2])) <= 0.75
+    assert np.mean(transform[:2, 2] - expected[:2, 2]) < 1.0
+
+    no_fixed_mask = np.zeros(shape=fixed_img.shape, dtype=int)
+    no_moving_mask = np.zeros(shape=moving_img.shape, dtype=int)
+    with pytest.raises(ValueError, match=r".*The foreground is missing in the mask.*"):
+        _ = prealignment(fixed_img, moving_img, no_fixed_mask, no_moving_mask)
+
+    with pytest.raises(
+        ValueError,
+        match=r".*Mismatch of shape between image and its corresponding mask.*",
+    ):
+        _ = prealignment(fixed_img, moving_img, moving_mask, fixed_mask)
+
+    with pytest.raises(
+        ValueError, match=r".*Please select the rotation step in between 10 and 20.*"
+    ):
+        _ = prealignment(
+            fixed_img, moving_img, fixed_mask, moving_mask, rotation_step=9
+        )
+
+    with pytest.raises(
+        ValueError, match=r".*Please select the rotation step in between 10 and 20.*"
+    ):
+        _ = prealignment(
+            fixed_img, moving_img, fixed_mask, moving_mask, rotation_step=21
+        )
+
+
+def test_dice_overlap_range():
+    """Test if the value of rotation step is within the range"""
+    fixed_img = np.random.randint(20, size=(256, 256))
+    moving_img = np.random.randint(20, size=(256, 256))
+    fixed_mask = np.random.randint(2, size=(256, 256))
+    moving_mask = np.random.randint(2, size=(256, 256))
+
+    _ = prealignment(fixed_img, moving_img, fixed_mask, moving_mask, dice_overlap=0.5)
```

Comment:
Please can you check the expected value here?

Reply:
I have removed this line since it was a redundant test. The expected value is checked here: 
https://github.com/TissueImageAnalytics/tiatoolbox/blob/4f038cfeb2ef3f4687c579c03c466f87be5bd326/tests/test_wsi_registration.py#L25-L37